src/devices/cpu/upd177x/upd177x.cpp

```diff
@@ -0,0 +1,797 @@
+// license:BSD-3-Clause
+// copyright-holders:Wilbert Pol
+/*
+
+TODO:
+- Only upd1771c supported
+- Noise, time, and external interrupts not implemented
+- CH1/CH2 not implemented
+- Not all instructions implemented
+- Are PNC1 and PNC2 updated every machine cycle or only when the noise counter overflows?
```

Comment:
They're updated on noise interrupt trigger: rising edge of selected noise counter bit.